automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala

```diff
@@ -64,63 +59,65 @@ class FreeTrialSpec extends FreeSpec with BeforeAndAfterEach with Matchers with
           }
-        }
 
-        // Verify that the user has been added to the corresponding billing project
-        val billingProject = Thurloe.keyValuePairs.getAll(subjectId).get("trialBillingProjectName")
-        assert(billingProject.nonEmpty, s"No trial billing project was allocated for the user ${testUser.email}.")
+          // Verify that the user has been added to the corresponding billing project
+          val billingProject = Thurloe.keyValuePairs.getAll(subjectId).get("trialBillingProjectName")
+          assert(billingProject.nonEmpty, s"No trial billing project was allocated for the user ${testUser.email}.")
 
-        val userBillingProjects = api.profile.getUserBillingProjects()(userAuthToken)
-        assert(userBillingProjects.nonEmpty, s"The trial user ${testUser.email} has no billing projects.")
+          val userBillingProjects = api.profile.getUserBillingProjects()(userAuthToken)
+          assert(userBillingProjects.nonEmpty, s"The trial user ${testUser.email} has no billing projects.")
 
-        val userHasTheRightBillingProject: Boolean = userBillingProjects.exists(_.values.toList.contains(billingProject.get))
-        assert(userHasTheRightBillingProject)
+          val userHasTheRightBillingProject: Boolean = userBillingProjects.exists(_.values.toList.contains(billingProject.get))
+          assert(userHasTheRightBillingProject)
 
-        // Verify that the user's project is removed from the account upon termination
-        val billingAccountUponEnrollment = Google.billing.getBillingProjectAccount(billingProject.get)(trialAuthToken)
-        assert(billingAccountUponEnrollment.nonEmpty, s"The user's project is not associated with a billing account.")
+          // Verify that the user's project is removed from the account upon termination
+          val billingAccountUponEnrollment = Google.billing.getBillingProjectAccount(billingProject.get)(trialAuthToken)
+          assert(billingAccountUponEnrollment.nonEmpty, s"The user's project is not associated with a billing account.")
 
-        api.trial.terminateUser(testUser.email)
+          api.trial.terminateUser(testUser.email)
 
-        val billingAccountUponTermination = Google.billing.getBillingProjectAccount(billingProject.get)(trialAuthToken)
-        val errMsg = "The trial user's billing project should have been removed from the billing account."
-        assert(billingAccountUponTermination.isEmpty, errMsg)
+          val billingAccountUponTermination = Google.billing.getBillingProjectAccount(billingProject.get)(trialAuthToken)
+          val errMsg = "The trial user's billing project should have been removed from the billing account."
+          assert(billingAccountUponTermination.isEmpty, errMsg)
 
-        registerCleanUpForDeleteTrialState()
+          // need to re-add a project owner in order for cleanup to succeed
+          addMembersToBillingProject(project, List(campaignManager.email), BillingProjectRole.Owner)(trialAuthToken)
```

Comment:
could you add this re-add-project-owner step to a cleanup step? It feels slightly awkward to have here.

Reply:
Is there a way to specify the order of cleanup operations? It's being called after the cleanup code from `withCleanBillingProject`. That code needs an admin user in order to release the project. If I could claim the billing project as the trial billing SA I think we could avoid this, but I don't know how to create `Credentials` for a service account.